src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1069,6 +1069,20 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return Promise.resolve(true);
 	}
 
+	/**
+	 * Replaces connection info uri with new uri.
+	 */
+	public changeConnectionUri(newUri: string, oldUri: string) {
+		this._connectionStatusManager.changeConnectionUri(newUri, oldUri);
+		if (!this._uriToProvider[oldUri]) {
+			this._logService.error(`No provider found for old URI : '${oldUri}'`);
+			throw new Error(nls.localize('connectionManagementService.noProviderForUri', 'Could not find provider for uri: {0}', oldUri));
+		}
+		// Provider will persist after disconnect, it is okay to overwrite the map if it exists from a previously deleted connection.
```

Comment:
Hm, not sure I like the behavior of the mapping persisting even after the editor is closed/disconnected. So something to keep an eye on if you continue to make changes here...we should ideally be cleaning up everything associated with a query once the editor is closed.

But it seems fine for this since that's existing behavior.